src/StatisticalAnalyser.java

```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
         this.graph = graph;
         this.regionsForVertex = regionsForVertex;
     }
-    // Link weight stats [0] - mean / [1] - standard deviation
-    public double[] linkWeightStats() {
-        double linksWeight = 0.0;
-        double[] stats = new double[2];
-        double mean = 0.0;
-        double std = 0.0;
+
+    public double linkWeightStdDeviation() {
+        double temp = 0.0;
+        double mean = linkWeightAverage();
 
         for (Edge link : graph.getEdges())
-            linksWeight += (double) link.weight();
+            temp += (link.weight() - mean)
+                    * (link.weight() - mean);
 
-        mean = linksWeight / (double) graph.getEdges().size();
-        stats[0] = mean;
+        int nEdges = graph.getEdges().size();
+        return  Math.sqrt(temp / (double) nEdges);
+    }
 
-        double temp = 0.0;
+    public double linkWeightAverage() {
+        double sum = 0.0;
         for (Edge link : graph.getEdges())
-            temp += ((double) link.weight() - mean)
-                    * ((double) link.weight() - mean);
+            sum += link.weight();
 
-        double variance = (temp / (double) (graph.getEdges().size()));
-        // size-1 for sample. We have population
+        int nEdges = graph.getEdges().size();
+        return sum / (double) nEdges;
+    }
 
-        std = Math.sqrt(variance);
-        stats[1] = std;
+    public int maxNumberOfRegions() {
+        return regSizes().get(regSizes().size() - 1);
+    }
 
-        return stats;
+    public int minNumberOfRegions() {
+        return regSizes().get(0);
     }
 
-    // Calculates the regions stats - [0] - Max / [1] - Min / [2] - Average
-    public double[] getRegionsStats() {
+    public double averageNumberOfRegions() {
+        int sum = 0;
+        for (int size : regSizes())
+            sum += size;
+        return (double)(sum / regSizes().size());
+    }
 
-        double[] stats = new double[3];
-        double average;
+    private List<Integer> regSizes() {
         List<Integer> regSizes = new ArrayList<>();
-
         for (Vertex r : graph.getVertices()) {
             regSizes.add(regionsForVertex.get(r).size());
         }
         Collections.sort(regSizes);
-
-        int sum = 0;
-        for (int size : regSizes) {
-            sum += size;
-        }
-        average = sum / regSizes.size();
-
-        stats[0] = (double) regSizes.get(regSizes.size() - 1);
-        stats[1] = (double) regSizes.get(0);
-        stats[2] = (double) average;
-        return stats;
+        return regSizes;
     }
 
     public double linkWeightMean(ArrayList<ArrayList<Path>> paths) {
```

Comment:
I didn't modify linkWeightMean and pathWeightMean methods that receives an A<A<Paths>>.